api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -502,6 +500,26 @@ private boolean userNeverLoggedIn(com.google.api.services.admin.directory.model.
     return ResponseEntity.status(HttpStatus.NO_CONTENT).build();
   }
 
+  @Override
+  public ResponseEntity<Void> updatePageVisits(PageVisit newPageVisit) {
+    User user = userProvider.get();
+    List<org.pmiops.workbench.db.model.PageVisit> oldPageVisits =
+      new ArrayList<>(user.getPageVisits());
+    Timestamp timestamp = new Timestamp(clock.instant().toEpochMilli());
+    boolean shouldAdd = oldPageVisits.stream().noneMatch(v -> v.getPageId().equals(newPageVisit.getPage()));
+    if (shouldAdd) {
+      org.pmiops.workbench.db.model.PageVisit firstPageVisit =
+        new org.pmiops.workbench.db.model.PageVisit();
+      firstPageVisit.setPageId(newPageVisit.getPage());
+      firstPageVisit.setUser(user);
+      firstPageVisit.setFirstVisit(timestamp);
+      oldPageVisits.add(firstPageVisit);
+      user.setPageVisits(new HashSet<>(oldPageVisits));
+      userDao.save(user);
+    }
```

Comment:
Two questions here.

1. Why create the list of oldPageVisits and timestamp outside of this block? We only need them in the case of shouldAdd == true
2. I asked in a previous comment, why do we need the temp list at all? Can't we just call user.getPageVisits().add()? Does that not work with our use of JPA?